simulator/src/bin/generate-example-screenshots.rs

```diff
@@ -0,0 +1,216 @@
+            //! <div style="flex-grow: 1;">
+//!
+//! ```rust
+//! # let mut display = embedded_graphics::mock_display::MockDisplay::default();
+//! {}
+//! # Ok::<(), core::convert::Infallible>(())
+//! ```
+//!
+//! </div>
+//! </div>
+//!"#,
+            $title,
+            $description,
+            $title,
+            screenshot,
+            docs.lines().collect::<Vec<_>>().join("\n//! ")
+        );
+    };
+}
+
+fn main() {
+    let output_base = "./target/drawing-ops";
+    std::fs::create_dir_all(output_base).expect("Output directory could not be created");
+
+    let mut display: SimulatorDisplay<pixelcolor::Rgb888> =
+        SimulatorDisplay::new(Size::new(64, 64));
+
+    op!(
+        display,
+        "Draw a single pixel",
+        "This example draws a single green pixel.",
+        {
+            use embedded_graphics::{pixelcolor::Rgb888, prelude::*};
+
+            Pixel(Point::new(32, 32), Rgb888::GREEN).draw(&mut display)?;
+        }
+    );
+
+    op!(
+        display,
+        "Draw a line",
+        "This example draws a red line with 1px stroke.",
+        {
+            use embedded_graphics::{
+                pixelcolor::Rgb888, prelude::*, primitives::Line, style::PrimitiveStyleBuilder,
+            };
+
+            Line::new(Point::new(16, 24), Point::new(48, 40))
+                .into_styled(
+                    PrimitiveStyleBuilder::new()
+                        .stroke_width(1)
+                        .stroke_color(Rgb888::RED)
+                        .build(),
+                )
+                .draw(&mut display)?;
+        }
+    );
+
+    op!(
+        display,
+        "Draw a rectangle",
+        "This example draws a rectangle with a 2px thick red stroke and cyan fill color.",
+        {
+            use embedded_graphics::{
+                pixelcolor::Rgb888, prelude::*, primitives::Rectangle, style::PrimitiveStyleBuilder,
+            };
+
+            Rectangle::new(Point::new(16, 24), Point::new(48, 40))
+                .into_styled(
+                    PrimitiveStyleBuilder::new()
+                        .stroke_width(2)
+                        .stroke_color(Rgb888::RED)
+                        .fill_color(Rgb888::CYAN)
+                        .build(),
+                )
+                .draw(&mut display)?;
+        }
+    );
+
+    op!(
+        display,
+        "Draw a circle",
+        "This example draws a circle with no stroke and a solid blue fill.",
+        {
+            use embedded_graphics::{
+                pixelcolor::Rgb888, prelude::*, primitives::Circle, style::PrimitiveStyleBuilder,
+            };
+
+            Circle::new(Point::new(22, 22), 20)
+                .into_styled(
+                    PrimitiveStyleBuilder::new()
+                        .fill_color(Rgb888::BLUE)
+                        .build(),
```

Comment:
Should we use `PrimitiveStyle::with_fill(Rgb888::BLUE)` instead? IMO this would be recommended for simple styles.